File: randomNameGenerator.Backend/randomNameGenerator/Models/Name.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace randomNameGenerator.Models
{
    public class Name
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public string Text { get; set; } = string.Empty;
    }
}

File: randomNameGenerator.Frontend/src/app/name.ts
export class Name {
    id: string = '';
    text: string = '';
    createdAt: Date = new Date();

    copyFrom(t: IName){
        this.id = t.id;
        this.text = t.text;
        this.createdAt = new Date(t.createdAt);
    }
}
export interface IName{
    id: string;
    text: string;
    createdAt: string;
}

